pocs/utils/rs232.py

```diff
@@ -135,28 +173,28 @@ def read(self, retry_limit=None, retry_delay=None):
                 break
             time.sleep(retry_delay)
             retry_limit -= 1
-
-        # self.logger.debug('Serial read: {}'.format(response_string))
         return response_string
 
     def get_reading(self):
         """Read a line and return the timestamp of the read.
 
         Returns:
-            str: Item in queue
+            A pair (tuple) of (timestamp, line). The timestamp is the time of completion of the
+            readline operation.
         """
-
-        # TODO(jamessynge): Consider collecting the time (utc?) after the read completes,
-        # so that long delays reading don't appear to have happened much earlier.
         try:
+            # Get the timestamp after the read so that a long delay on reading doesn't make it
+            # appear that the read happened much earlier than it did.
+            line = self.read()
             ts = time.strftime('%Y-%m-%dT%H:%M:%S %Z', time.gmtime())
             info = (ts, self.read())
```

Comment:
Should this be `line` instead of another `self.read()`?

Reply:
Doh! Test added that detected this bug, and bug fixed.